External/Plugins/HaXeContext/Context.cs

```diff
@@ -903,65 +903,143 @@ public override bool IsImported(MemberModel member, int atLine)
+                                        break;
+                                    }
+                                }
+                                if (!found && className == "Function") result = stubFunctionClass;
+                            }
+                        }
+                        if (result.IsVoid()) result = GetModel(package, className, inPackage);
+                    }
+                }
             }
-
-            // typed array
-            if (cname.IndexOf('@') > 0)
-                return ResolveTypeIndex(cname, inFile);
-
-            string package = "";
-            string inPackage = (features.hasPackages && inFile != null) ? inFile.Package : "";
-
-            int p = cname.LastIndexOf('.');
-            if (p > 0)
+            if (!result.IsVoid() && resolveStaticExtensions && inFile?.Imports.Count > 0)
             {
-                package = cname.Substring(0, p);
-                cname = cname.Substring(p + 1);
+                var extensions = ResolveStaticExtensions(result, inFile);
+                if (extensions.Count > 0)
+                {
+                    result = (ClassModel) result.Clone();
+                    result.Members.Merge(extensions);
+                }
             }
-            else 
-            {
-                // search in file
-                if (inFile != null)
-                    foreach (ClassModel aClass in inFile.Classes)
-                        if (aClass.Name == cname)
-                            return aClass;
+            return result;
+        }
 
-                // search in imported classes
-                var found = false;
-                MemberList imports = ResolveImports(inFile);
-                foreach (MemberModel import in imports)
+        /// <summary>
+        /// https://haxe.org/manual/lf-static-extension.html
+        /// </summary>
+        /// <param name="target"></param>
+        /// <param name="inFile">Current file</param>
+        /// <returns></returns>
+        public MemberList ResolveStaticExtensions(ClassModel target, FileModel inFile)
+        {
+            var result = new MemberList();
+            IEnumerable<MemberModel> imports = inFile.Imports.Items;
+            if (GetCurrentSDKVersion() >= "3.3.0") imports = ResolveDefaults(inFile.Package).Items.Concat(imports);
+            var importModels = imports.Where(it => it.Flags.HasFlag(FlagType.Using)).ToArray();
+            for (var i = importModels.Length - 1; i >= 0; i--)
+            {
+                var import = importModels[i];
+                var type = ResolveType(import.Name, inFile, false);
+                if (type.IsVoid() || type.Members.Count == 0) continue;
+                var access = TypesAffinity(target, type);
+                var extends = target;
+                while (!extends.IsVoid())
                 {
-                    if (import.Name == cname)
+                    foreach (MemberModel member in type.Members)
```

Comment:
Is that something that potentially could be part of the `completionCache`? So at this point we only iterate over the potential static extension and select the ones compatible.